docs/user-guide/ffi.md

```diff
@@ -0,0 +1,122 @@
+# PyScript FFI
+
+The reason we decided to incrementally provide a unified *pyscript.ffi* utility is that [Pyodide](https://pyodide.org/en/stable/usage/api/python-api/ffi.html)'s *ffi* is only partially implemented in *MicroPython* but there are fundamentals differences even with the few common utilities both projects provide, hence our intention to smooth out their usage with all the features or caveats one needs to be aware of.
+
+Our `pyscript.ffi` offers the following utilities:
+
+  * `ffi.to_js(reference)` to convert any Python reference to its *JS* counterpart
+  * `ffi.create_proxy(def_or_lambda)` to proxy a generic Python's function into a *JS* one, without destroying its reference right away
+
+## to_js
+
+In [Pyodide project](https://pyodide.org/en/stable/usage/api/python-api/ffi.html#pyodide.ffi.to_js), this utility converts by default Python dictionaries into a *Map*, or prefer the *Map* indirection as the *Map* somehow better reflects the `obj.get(field)` native Python way to retrieve a value from a dictionary.
+
+However, we have noticed that this default conversion plays very badly with pretty much any native or user made *JS* API, so that most of the code around `to_js` needs to explicitly define a `dict_converter=js.Object.fromEntries` instead of mapping Python's dictionaries directly as JS' objects literal.
+
+On top of that, in *MicroPython* the default conversion already produces *object literals*, making the need to specify a different converter pretty pointless.
+
+As we all known though, *explicit is better than implicit*, so whenever an object literal is expected and no reason to hold a Python reference in the JS world is needed, using `to_js(python_dictionary)` will guarantee the entity to be copied or passed as *object literal* to the consumer of that dictionary.
+
+```html title="to_js: pyodide.ffi VS pyscript.ffi"
+<!-- works on Pyodide (type py) only -->
+<script type="py">
+  from pyodide.ffi import to_js
+
+  # default into JS new Map([["a", 1], ["b", 2]])
+  to_js({"a": 1, "b": 2})
+</script>
+
+<!-- works on both Pyodide and MicroPython -->
+<script type="py">
+  from pyscript.ffi import to_js
+
+  # always default into JS {"a": 1, "b": 2}
+  to_js({"a": 1, "b": 2})
+</script>
+```
+
+!!! Note
+
+    It is still possible to specify a different `dict_converter` or use Pyodide
+    specific features while converting Python references by simply overriding
+    the explicit field for `dict_converter`. However, we cannot guarantee
+    all fields and features provided by Pyodide will work the same on MicroPython.
+
+## create_proxy
+
+In [Pyodide project](https://pyodide.org/en/stable/usage/api/python-api/ffi.html#pyodide.ffi.create_proxy), this utility guarantees that a Python lambda or callback won't get Garbage Collected immediately after.
```

Comment:
* "In Pyodide project" -> "In the Pyodide project"
* "won't get Garbage Collected immediately after." -> "won't be garbage collected immediately."